packages/aws-cdk/test/account-cache.test.ts

```diff
@@ -89,3 +110,28 @@ test(`cache is nuked if it exceeds ${AccountAccessKeyCache.MAX_ENTRIES} entries`
     await nukeCache(cacheDir);
   }
 });
+
+function withMocked<A extends object, K extends keyof A, B>(obj: A, key: K, block: (fn: jest.Mocked<A>[K]) => B): B {
```

Comment:
this is fun :)

Reply:
And maybe a little overkill, but may come in useful in other cases as well... remember that this exists (somewhere) !~